From 066189add979d2e4c74463592e5021bd060d5a51 Mon Sep 17 00:00:00 2001
From: Mohammed Rafi KC <rkavunga@redhat.com>
Date: Thu, 11 Jul 2019 12:46:47 +0530
Subject: [PATCH 241/255] Revert "glusterd/svc: glusterd_svcs_stop should call
 individual wrapper function"

This reverts commit 79fff98f9ca5f815cf0227312b9a997d555dad29.

BUG: 1471742
Change-Id: I258040ed9be6bc3b4498c76ed51d59258c55acff
Signed-off-by: Mohammed Rafi KC <rkavunga@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/175950
Tested-by: RHGS Build Bot <nigelb@redhat.com>
Reviewed-by: Sunil Kumar Heggodu Gopala Acharya <sheggodu@redhat.com>
---
 xlators/mgmt/glusterd/src/glusterd-shd-svc.c    | 12 ++----------
 xlators/mgmt/glusterd/src/glusterd-svc-helper.c | 10 +++++-----
 2 files changed, 7 insertions(+), 15 deletions(-)

diff --git a/xlators/mgmt/glusterd/src/glusterd-shd-svc.c b/xlators/mgmt/glusterd/src/glusterd-shd-svc.c
index 981cc87..75f9a07 100644
--- a/xlators/mgmt/glusterd/src/glusterd-shd-svc.c
+++ b/xlators/mgmt/glusterd/src/glusterd-shd-svc.c
@@ -656,18 +656,10 @@ glusterd_shdsvc_stop(glusterd_svc_t *svc, int sig)
     int pid = -1;
 
     conf = THIS->private;
-    GF_VALIDATE_OR_GOTO("glusterd", conf, out);
     GF_VALIDATE_OR_GOTO("glusterd", svc, out);
     svc_proc = svc->svc_proc;
-    if (!svc_proc) {
-        /*
-         * This can happen when stop was called on a volume that is not shd
-         * compatible.
-         */
-        gf_msg_debug("glusterd", 0, "svc_proc is null, ie shd already stopped");
-        ret = 0;
-        goto out;
-    }
+    GF_VALIDATE_OR_GOTO("glusterd", svc_proc, out);
+    GF_VALIDATE_OR_GOTO("glusterd", conf, out);
 
     /* Get volinfo->shd from svc object */
     shd = cds_list_entry(svc, glusterd_shdsvc_t, svc);
diff --git a/xlators/mgmt/glusterd/src/glusterd-svc-helper.c b/xlators/mgmt/glusterd/src/glusterd-svc-helper.c
index 6a3ca52..f7be394 100644
--- a/xlators/mgmt/glusterd/src/glusterd-svc-helper.c
+++ b/xlators/mgmt/glusterd/src/glusterd-svc-helper.c
@@ -86,25 +86,25 @@ glusterd_svcs_stop(glusterd_volinfo_t *volinfo)
     priv = this->private;
     GF_ASSERT(priv);
 
-    ret = priv->nfs_svc.stop(&(priv->nfs_svc), SIGKILL);
+    ret = glusterd_svc_stop(&(priv->nfs_svc), SIGKILL);
     if (ret)
         goto out;
 
-    ret = priv->quotad_svc.stop(&(priv->quotad_svc), SIGTERM);
+    ret = glusterd_svc_stop(&(priv->quotad_svc), SIGTERM);
     if (ret)
         goto out;
 
     if (volinfo) {
-        ret = volinfo->shd.svc.stop(&(volinfo->shd.svc), SIGTERM);
+        ret = glusterd_svc_stop(&(volinfo->shd.svc), PROC_START_NO_WAIT);
         if (ret)
             goto out;
     }
 
-    ret = priv->bitd_svc.stop(&(priv->bitd_svc), SIGTERM);
+    ret = glusterd_svc_stop(&(priv->bitd_svc), SIGTERM);
     if (ret)
         goto out;
 
-    ret = priv->scrub_svc.stop(&(priv->scrub_svc), SIGTERM);
+    ret = glusterd_svc_stop(&(priv->scrub_svc), SIGTERM);
 out:
     return ret;
 }
-- 
1.8.3.1

